main/mio.c

```diff
@@ -999,8 +1004,10 @@ char *mio_gets (MIO *mio, char *s, size_t size)
 				s[i] = 0;
 				rv = s;
 			}
-			if (mio->impl.mem.pos >= mio->impl.mem.size)
+			if (!newline && pos >= buf_size)
```

Comment:
@b4n That's why I introduced the change - it _didn't_ match the FILE implementation - see the comment or just try it. If you have a file with:

```
abcd\nEOF
```

and do gets() on it, it should first read `abcd\n` without setting EOF (because EOF hasn't been reached yet) and set EOF only on the subsequent call.